tools/plugin-toolkit/src/index.ts

```diff
@@ -1,3 +1,62 @@
-// (C) 2019-2021 GoodData Corporation
+// (C) 2021 GoodData Corporation
 
-export { mySdkFunction } from "./code/code";
+import { Command, program } from "commander";
+import * as pkg from "../package.json";
+import { addPluginCmdAction } from "./addPluginCmd";
+import { initCmdAction } from "./initCmd";
+import { usePluginCmdAction } from "./usePluginCmd";
+
+program
+    .version(pkg.version)
+    .name("GoodData Plugin Development Toolkit")
+    .option("--accept-untrusted-ssl", "Allows to run the tool with host, that has untrusted ssl certificate")
+    .option("--hostname <hostname>", "URL of your GoodData host")
+    .option("-c, --config <config>", "path to configuration file");
+
+const initCmd: Command = program
+    .command("init-dashboard")
+    .description("Initialize a new dashboard plugin")
+    .argument("[plugin-name]", "Name of the plugin to create")
+    .option("--target-dir <path>", 'Path to the directory to create the plugin in (default: ".")')
+    .option("--no-install", "Skip yarn installing the plugin dependencies")
+    .option("--verbose", "Output additional logs, useful mainly for debugging and bug reports")
+    .option("--backend <backend>", "Setting backend of the app (default: bear backend)")
+    .option("--flavor <flavor>", "Language flavor of the plugin, either TypeScript (ts) or JavaScript (js)")
+    .action(async (pluginName) => {
+        return initCmdAction(pluginName, {
+            programOpts: program.opts(),
+            commandOpts: initCmd.opts(),
+        });
+    });
+
+const addPluginCmd: Command = program
+    .command("add-plugin")
+    .description("Add plugin to a workspace so that it can be used on the dashboards in that workspace.")
+    .option("--username <email>", "Your username that you use to log in to GoodData platform.")
+    .option("--workspace-id <id>", "Workspace id to which you want to add the plugin.")
+    .action(async () => {
+        return addPluginCmdAction({
+            programOpts: program.opts(),
+            commandOpts: addPluginCmd.opts(),
+        });
+    });
+
+const usePluginCmd: Command = program
+    .command("use-plugin")
+    .argument("[plugin-id]", "Plugin id which you want to use on the dashboard.")
+    .option("--username <email>", "Your username that you use to log in to GoodData platform.")
+    .option("--workspace-id <id>", "Workspace id that contains dashboard that should use the plugin.")
+    .option("--dashboard-id <id>", "Dashboard id on which you want to use the plugin.")
```

Comment:
I wonder if the dashboardId should be a non-named variadic argument. So that you could set up more dashboards at once? Or maybe even make this accept stdin so that you can pipe to it?

Reply:
In other words, make dashId an argument and the plugin id an option? Because you will probably set one plugin to more dashboards not the other way around right?

Reply:
good point; i'll ponder that, see what i can come up with. things will get cleaner once i do the add plugin